ipa-core/src/report/hybrid.rs

```diff
@@ -333,33 +334,39 @@ where
     /// # Panics
     /// If report length does not fit in `u16`.
     pub fn encrypted_len(&self) -> u16 {
-        self.ciphertext_len() + u16::try_from(self.info.byte_len()).unwrap()
+        self.ciphertext_len()
     }
 
     /// # Errors
     /// If there is a problem encrypting the report.
+    /// # Panics
+    /// If info length + report length does not fit in `u16`.
     pub fn delimited_encrypt_to<R: CryptoRng + RngCore, B: BufMut>(
         &self,
         key_id: KeyIdentifier,
         key_registry: &impl PublicKeyRegistry,
+        info: &HybridConversionInfo,
         rng: &mut R,
         out: &mut B,
     ) -> Result<(), InvalidHybridReportError> {
-        out.put_u16_le(self.encrypted_len());
-        self.encrypt_to(key_id, key_registry, rng, out)
+        out.put_u16_le(self.encrypted_len() + u16::try_from(info.byte_len()).unwrap());
+        self.encrypt_to(key_id, key_registry, info, rng, out)
     }
 
     /// # Errors
     /// If there is a problem encrypting the report.
+    /// # Panics
+    /// If info length + report length does not fit in `u16`.
     pub fn encrypt<R: CryptoRng + RngCore>(
         &self,
         key_id: KeyIdentifier,
         key_registry: &impl PublicKeyRegistry,
+        info: &HybridConversionInfo,
         rng: &mut R,
     ) -> Result<Vec<u8>, InvalidHybridReportError> {
-        let mut out = Vec::with_capacity(usize::from(self.encrypted_len()));
-        self.encrypt_to(key_id, key_registry, rng, &mut out)?;
-        debug_assert_eq!(out.len(), usize::from(self.encrypted_len()));
+        let mut out = Vec::with_capacity(usize::from(self.ciphertext_len() + u16::try_from(info.byte_len()).unwrap()));
```

Comment:
there seems to be a fair amount of similarities between this code and the code above. Any way to reduce the copy-paste?